components/brave_extension/extension/brave_extension/content_cosmetic.ts

```diff
@@ -183,17 +183,20 @@ const isFirstPartyUrl = (url: string): boolean => {
   }
 
   const parsedTargetDomain = getParsedDomain(url)
-  if (!parsedTargetDomain) {
-    // If we cannot determine the party-ness of the resource,
-    // consider it first-party.
-    console.debug(`Cosmetic filtering: Unable to determine party-ness of "${url}"`)
+  if (parsedTargetDomain.type !== _parsedCurrentDomain.type) {
+    return false
+  } else if (parsedTargetDomain.type === ParseResultType.Listed) {
```

Comment:
small nits here, but can we change these mega OR statements to more clearly describe the policy they're enforcing.  So, for example:
```
const isSameEtldP1 = (_parsedCurrentDomain.icann.topLevelDomains === parsedTargetDomain.icann.topLevelDomains &&
      _parsedCurrentDomain.icann.domain === parsedTargetDomain.icann.domain)
return isSameEtldP1
```

Or 
```
const looksLikeLocalHostOrigin = (parsedTargetDomain.type === ParseResultType.NotListed
      || parsedTargetDomain.type === ParseResultType.Ip
      || parsedTargetDomain.hostname === ParseResultType.Reserved)

if (looksLikeLocalHostOrigin) {
  return _parsedCurrentDomain.hostname === parsedTargetDomain.hostname
}
```

Otherwise, its pretty difficult to understand the "why"